code/modules/admin/verbs/playsound.dm

```diff
@@ -119,7 +119,16 @@
 					music_extra_data["upload_date"] = data["upload_date"]
 					music_extra_data["album"] = data["album"]
 
-					var/res = tgui_alert(usr, "Show the title of and link to this song to the players?\n[title]",, list("No", "Yes", "Cancel"))
+					var/duration = null
+					if (data["duration"] * 1 SECONDS > 10 MINUTES)
+						duration = tgui_alert(usr, "This song is over 10 minutes long. Are you sure you want to play it?", "Length Warning!", list("No", "Yes", "Cancel"))
+					switch(duration)
+						if("No")
+							return
+						if("Cancel")
+							return
```

Comment:
```suggestion
					switch(duration)
						if("No", "Cancel", null)
							return
```

Closing out of a `tgui_alert` will result in a null value, which _must_ be respected. i note some other of the code doesn't really do this, which is sad, but let's not learn from mistakes.